### PR TITLE
[SDK] Add support for transaction on deploy

### PR DESCRIPTION
**Pull request by adam-maj**

### CODE REVIEW

File: packages/sdk/src/evm/core/classes/transactions.ts [hunk 1]
```diff
@@ -694,3 +723,129 @@ export class Transaction<TResult = TransactionResult> {
...
+    } catch (err) {
+      // If tx.wait() fails, it just gives us a generic "transaction failed"
+      // error. So instead, we need to call static to get an informative error message
+      await this.simulate();
+
+      // If transaction simulation (static call) doesn't throw, then throw with the message that we have
+      throw await this.deployError(err);
+    }
+
+    return utils.getContractAddress({ from: tx.from, nonce: tx.nonce });
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/core/classes/transactions.ts:787]: this is neat - does it parse the tx log for that address + nonce?

Comment (adam-maj) [reply packages/sdk/src/evm/core/classes/transactions.ts:787]: It actually just does this `getAddress(hexDataSlice(keccak256(encode([ from, nonce ])), 12))` - which ig is the way to get the address for a standard direct deploy

**Review by joaquim-verges — APPROVED** [review]
looks good to me - did you test this on the dashboard? can test based off the cwv2 branch

**Comment by adam-maj** [comment]
Tested on dashboard - lgtm

**Pull request merged**